controller/scheduler/scheduler.go

```diff
@@ -256,11 +256,11 @@ func (s *Scheduler) Run() error {
 		return err
 	}
 
-	var err error
-	s.isLeader, err = s.discoverd.Register()
+	isLeader, err := s.discoverd.Register()
 	if err != nil {
 		return err
 	}
+	s.HandleLeaderChange(isLeader)
```

Comment:
OK so right now the `rectifyAll` call will be a harmless no-op. I guess we'll let the formation stream worry about what to rectify.